### ISSUE TITLE
RequiresDuplicateDetection can lead to a message loss in v6

### ISSUE THREAD

**Issue by SeanFeldman** [Particular/NServiceBus.AzureServiceBus#242]
## Description

This is a separate issue for identical [issue raised for v7 of ASB](https://github.com/Particular/NServiceBus.AzureServiceBus/issues/201)

ASB transport allows [native duplicates detection](http://docs.particular.net/nservicebus/azure-service-bus/configuration/configuration#controlling-entities-queues). By default it's set to false, but can be enabled via [transport configuration](http://docs.particular.net/nservicebus/azure-service-bus/configuration/azureservicebusqueueconfig) or directly on the entity. 

_When_ `RequiresDuplicateDetection` is enabled in conjunction with SLR feature, it could lead to a message loss (see details in the [original issue](https://github.com/Particular/NServiceBus.AzureServiceBus/issues/201)). 

In addition to that, without SLR, but with a significant time windows  _and_ SC involved retried messages could be lost as well. (Note: Duplicate detection time window can be set up up to 7)

## TL;DR 

2 scenarios with `RequiresDuplicateDetection` enabled when messages can be lost:

1. SLR enabled
1. FLR + SC combo

## Issue

While it's an edge case, it still qualifies for a back-ported fix. The [question raised](https://github.com/Particular/NServiceBus.AzureServiceBus/issues/201#issuecomment-222603142) was:
> can we back-port a fix that isn't going to cause backwards compatibility issues

To answer this question, we'd need to understand where transport message Id that doesn't have the value of the NSB Message Id is possibly going to break things, for Core, Particular Platform, and custom customer code (_if_ we can determine that).

@Particular/azure-service-bus-maintainers @andreasohlund Please chime in.

**Comment by yvesgoeleven** [comment]
@SeanFeldman @andreasohlund  As mentioned earlier today, I don't think any of our customers are running with native dedupe on, and no one has reported such an occurance so far. So I don't think that the benefits of backporting outweigh the potential problems that may occur by changing the message id

**Comment by SeanFeldman** [comment]
We can't backport this as we don't know what the affect on production will be.
Fully addressed in v7 of the ASB.
